src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -246,6 +246,27 @@ public function testCreate()
         $this->assertEquals('http://test.com/foo', $request->getUri());
     }
 
+    public function testIndexDotPhpPathInfo()
+    {
+        // assume no rewrite rule: /index.php --> /
+        // assume index.php is a real file.
+        // assume rewrite rule passes all requests not referring directly to
+        // files in the filesystem to index.php.
+        $request = Request::create('http://test.com/index.php.php', 'GET', [], [], [],
+            [
+                'DOCUMENT_ROOT' => '/var/www/www.test.com',
+                'SCRIPT_FILENAME' => '/var/www/www.test.com/index.php',
+                'SCRIPT_NAME' => '/index.php',
+                'PHP_SELF' => '/index.php',
+            ]);
+        $this->assertEquals('http://test.com/index.php/.php', $request->getUri());
```

Comment:
that case looks wrong to me though, as that would be a different Uri that the input

Reply:
What should the uri be though? If you make a request to index.php/something or /something in this situation you'd want the path info to be /something. And it is. I think this is a special edge case because of the way htaccess rules work for Drupal (and probably a number of other projects). The relevant parts of our rule is:
```
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteCond %{REQUEST_URI} !=/favicon.ico
  RewriteRule ^ index.php [L]
```
I think we should treat a request to index.phpSOMETHING as a request to SOMETHING - regardless of whether SOMETHING starts with a /... which means we need to duplicate the logic of when there is no base path...

This is just above the code I've added.
```
        if ('' !== $requestUri && '/' !== $requestUri[0]) {
            $requestUri = '/'.$requestUri;
        }

        if (null === ($baseUrl = $this->getBaseUrl())) {
            return $requestUri;
        }
```

Reply:
For `index.php/something`, It totally agree. But this is **not** what this test is covering.